src/Raven.Server/Documents/Handlers/Processors/OngoingTasks/AbstractOngoingTasksHandlerProcessorForDeleteOngoingTask.cs

```diff
@@ -120,6 +127,17 @@ public async Task Complete(string raftRequestId)
                         await _requestHandler.WaitForIndexNotificationAsync(index);
                     }
                 }
+
+                if (_deletingQueueSink.Name != null)
+                {
+                    foreach (var script in _deletingQueueSink.Scripts)
+                    {
+                        var (index, _) = await _serverStore.RemoveQueueSinkProcessState(_context, _requestHandler.DatabaseName, _deletingQueueSink.Name, script,
+                            $"{raftRequestId}/{script}");
+
+                        await _requestHandler.WaitForIndexNotificationAsync(index);
```

Comment:
shouldn't we just gather in this method max Index and wait once at the end?

Reply:
done

Reply:
thought taht you will also handle `_deletingEtl`, similar thing there? one maxIndex can be used?